chain/x/zoracle/internal/types/data_source.go

```diff
@@ -0,0 +1,23 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// DataSource is a type to store detail of data source
+type DataSource struct {
+	Owner      sdk.AccAddress `json:"owner"`
+	Name       string         `json:"name"`
+	Fee        sdk.Coins      `json:"fee"`
+	Executable []byte         `json:"executable"`
+}
+
+// NewDataSource - contructor of DataSource struct
```

Comment:
Let's follow the pattern that we usually do. NewDataSource creates a new DataSource instance.